lib/ssh_scan/api.rb

```diff
@@ -162,8 +169,25 @@ class API < Sinatra::Base
         if worker_id.empty? || uuid.empty?
           return {"accepted" => "false"}.to_json
         end
+        results = JSON.parse(request.body.first).first
+        settings.current_avg = settings.current_avg +
+        (results['scan_duration_seconds'] - settings.current_avg) / settings.count_req
```

Comment:
Oh, I thought `results['scan_duration_seconds'] - settings.current_avg` is equal to `time_spent_on_curr_task`, whereas `results['scan_duration_seconds']` is.

Then, its correct. (y)